code/modules/reagents/reagent_containers.dm

```diff
@@ -67,7 +67,8 @@
 	return
 */
 /obj/item/weapon/reagent_containers/afterattack(obj/target, mob/user , flag)
-	return
+	if(user.a_intent == "harm")
+		return ..()

```

Comment:
Yes, but the "bash head in with potato" is completely overridden behavior - it does not call attacked_by. See the first link.